### PR TITLE
adding internal jwt generation endpoint for turkeyorch ops

### PR DESCRIPTION
**Pull request by tanfarming**
note: `/api-internal` endpoints are unreachable from public internet on turkey clusters, they also requires an turkey cluster scoped api key

### CODE REVIEW

File: lib/ret_web/controllers/api-internal/v1/AuthTokenController.ex [hunk 1]
```diff
@@ -0,0 +1,18 @@
+defmodule RetWeb.ApiInternal.V1.AuthTokenController do
+  require Logger
+
+  use RetWeb, :controller
+
+  def post(conn, %{"email" => email}) when is_binary(email) do
+    {:ok, token, _params} =
+      if !System.get_env("TURKEY_MODE") do
```
Comment (bryanenders) [inline lib/ret_web/controllers/api-internal/v1/AuthTokenController.ex:8]: It’s best-practice to quarantine all `System.get_env/1` calls.
```suggestion
      if not Ret.turkey_mode?() do
```

**lib/ret/ret.ex**
```elixir
@spec turkey_mode? :: boolean
def turkey_mode?,
  do: Application.fetch_env!(:ret, :turkey_mode?)
```

**config/config.exs**
```elixir
config :ret,
  ecto_repos: [Ret.Repo, Ret.SessionLockRepo],
  turkey_mode?: System.get_env("TURKEY_MODE") === "1"
```

**Review by keianhzo — APPROVED** [review]

**Review by bryanenders — CHANGES_REQUESTED** [review]

File: lib/ret_web/controllers/api-internal/v1/AuthTokenController.ex (hunk 1 above)
Comment (tanfarming) [reply lib/ret_web/controllers/api-internal/v1/AuthTokenController.ex]: thx for the code lol
commited

Comment (bryanenders) [reply lib/ret_web/controllers/api-internal/v1/AuthTokenController.ex]: No problem! You’ll want to make sure to update `ret.ex` and `config.exs` alongside `AuthTokenController.ex`.

File: config/config.exs [hunk 2]
```diff
@@ -60,6 +60,10 @@ config :peerage, log_results: false
 
 config :statix, prefix: "ret"
 
+config :ret,
+  ecto_repos: [Ret.Repo, Ret.SessionLockRepo],
+  turkey_mode?: System.get_env("TURKEY_MODE") === "1"
+
```
Comment (bryanenders) [inline config/config.exs:66]: Could you move this up to replace line 9? While this technically works, it overwrites the effects of line 9 and could be a pain-in-the-butt to find this far down.

File: lib/ret.ex [hunk 3]
```diff
@@ -40,4 +40,8 @@ defmodule Ret do
   defp valid_email_address?(string) do
     string =~ ~r/\S+@\S+/
   end
+
+  @spec turkey_mode? :: boolean
+  def turkey_mode?,
+    do: Application.fetch_env!(:ret, :turkey_mode?)
```
Comment (bryanenders) [inline lib/ret.ex:46]: This is a nitpick, but would you mind moving this function before `valid_email_address?/1` so these public functions are in alphabetical order?

File: config/config.exs (hunk 2 above)
Comment (tanfarming) [reply config/config.exs]: done

File: lib/ret.ex (hunk 3 above)
Comment (tanfarming) [reply lib/ret.ex:42]: done

**Review by bryanenders — APPROVED** [review]

File: lib/ret.ex (hunk 3 above)
Comment (bryanenders) [reply lib/ret.ex:42]: Thank you!

**Comment by bryanenders** [comment]
The lint-and-test failure should be solved with a quick **$** `mix format`

**Comment by tanfarming** [comment]
> The lint-and-test failure should be solved with a quick **$** `mix format`

yeah i ran that but my elixir version was v1.12 where our linter agent's on v1.8
i downgraded mine to 1.10 (oldest still available) and it worked
so there's some breaking changes on their linter somewhere around v1.11
i've also experienced a bunch of breaking changes when i was trying to move us out of distillery with latest elixir versions (was trying on v1.14) -- do you know if this is a pattern with elixir? if so maybe we want to sit on our current version for as long as possible, given that we are now using the cloudflare worker proxy across HC and turkey and are going to redeisgn ret's file system, we could potentially wait longer / jump to a higher version

**Comment by tanfarming** [comment]
ok this `quarantine all System.get_env/1 call` broke the feature lol
i'm guessing the evaluation's happening on build/compile time rather than runtime?
@bryanenders

**Comment by bryanenders** [comment]
There was some churn when the Elixir formatter was new. I haven’t seen changes for awhile, so I don’t expect that to be much of an issue after we upgrade.

**Comment by tanfarming** [comment]
> @tanfarming That makes sense. With distillery all the app config is being baked at compile time
yeah i forogt that runtime.exs wasn't a thing wtih elixir before v1.10, that's actually the reason why distillery's used

> until we can move to Elixir releases.
lets get togheter after EA and see if we can finally get this done. It's a pretty messy job, inaddition to fighting through the dependency hell, update ret codes against elixir's breaking changes, both HC and turkey's config and infra will need to be updated/redesigned against this change

File: config/config.exs [hunk 4]
```diff
@@ -6,7 +6,10 @@
 use Mix.Config
 
 # General application configuration
-config :ret, ecto_repos: [Ret.Repo, Ret.SessionLockRepo]
+
+config :ret,
+  ecto_repos: [Ret.Repo, Ret.SessionLockRepo],
+  turkey_mode?: System.get_env("TURKEY_MODE") === "1"
```
Comment (bryanenders) [inline config/config.exs:12]: You’ll want to revert this as well

File: lib/ret.ex [hunk 5]
```diff
@@ -37,6 +37,10 @@ defmodule Ret do
     end
   end
 
+  @spec turkey_mode? :: boolean
+  def turkey_mode?,
+    do: Application.fetch_env!(:ret, :turkey_mode?)
+
```
Comment (bryanenders) [inline lib/ret.ex:43]: And this

Comment (bryanenders) [reply lib/ret.ex:43]: Sorry for the hassle!

**Pull request merged**